tests/test_ext_commands.py

```diff
@@ -134,6 +137,31 @@ def test_sanitize_repo_commits_non_processed_files_to_target_branch(
     )
 
 
+def test_sanitize_repo_returns_fail_when_repo_has_staged_changes(
+    sanitizer_config, fake_repo
+):
+    tracked_file = fake_repo.file_infos[0].abspath
+    tracked_file.write_text("this is the new text!")
+    fake_repo.repo.git.add(tracked_file)
+
+    result = execute_sanitize_repo(
+        f"--file-list {fake_repo.file_list_path} "
+        f"--repo-root {fake_repo.path} "
+        "--no-commit"
+    )
+
+    assert result.status == plug.Status.ERROR
```

Comment:
This test is very nice and clean, but we may want to add a check on the message to make sure that we're actually getting the correct error.

```suggestion
def test_sanitize_repo_returns_fail_when_repo_has_staged_changes(
    sanitizer_config, fake_repo
):
    tracked_file = fake_repo.file_infos[0].abspath
    tracked_file.write_text("this is the new text!")
    fake_repo.repo.git.add(tracked_file)

    result = execute_sanitize_repo(
        f"--file-list {fake_repo.file_list_path} "
        f"--repo-root {fake_repo.path} "
        "--no-commit"
    )

    assert result.status == plug.Status.ERROR
    assert "uncommitted staged files" in result.msg
```